kong/db/dao/init.lua

```diff
@@ -352,14 +361,22 @@ function DAO:delete(primary_key)
     return nil, tostring(err_t), err_t
   end
 
+  local entity, err, err_t = self:select(primary_key)
+  if err then
+    return nil, err, err_t
+  end
+  if not entity then
+    return 0
```

Comment:
Our rationale was to not return the number of affected rows from `delete()` (because of the lack of support from Cassandra for this). Is the rationale behind this return value to indicate that no row was deleted?

Reply:
Yes. With the read-before-delete you know which rows will (should) be affected by a delete. The `0` means "I could not find any rows". I can replace it with a `true` if you prefer.